src/language-js/embed.js

```diff
@@ -16,6 +17,35 @@ const {
   utils: { mapDoc, stripTrailingHardline }
 } = require("../doc");
 
+const cssPlaceholder = new Placeholder("prettier");
+const CSS_PROPERTY_PLACEHOLDER = new Placeholder("prettier").get(0);
+const CSS_PRETTIER_IGNORE_PLACEHOLDER = new Placeholder("prettier").get(0);
+const CSS_EXTRA_SEMICOLON_MARK = new Placeholder("prettier").get(0);
+const placeholderPiecesToStringArray = pieces =>
+  pieces.map(({ isPlaceholder, string, placeholder }) =>
+    isPlaceholder ? placeholder : string
+  );
+const removeCSSComments = string =>
+  string
+    .replace(/\/\*\s*prettier-ignore\s*\*\//g, CSS_PRETTIER_IGNORE_PLACEHOLDER)
+    .replace(/\/\/\s*prettier-ignore/g, CSS_PRETTIER_IGNORE_PLACEHOLDER)
+    .replace(/\/\*[\s\S]*?\*\//g, "")
+    .replace(/\/\/.*/g, "")
```

Comment:
Can this accidentally remove things inside strings? Such as `a[href="https://foo"]`, `a { content: "/* prettier-ignore */" }`?

Reply:
could be, I'll check later.

FYI: this function mean to keep `prettier-ignore`, and only used to check some before after string. logic on L114-L142

Reply:
you are right, I  break [this case](https://deploy-preview-6794--prettier.netlify.com/playground/#N4Igxg9gdgLgprEAucBnVADAOlA2gCwCc4AzAXixAHoqSIJKBdAAmGYBJg4APAB0IC+AbgE4MIADQgIvGAEtoqZKACGhQhADuABTUIlKFQDcIcgCaSQAI0IqwAazgwAyrztyoAc2QxCAVzgpfBgAWwAbAHV8OXhUNzA4Z30YuSMYgE9kNCUpD1Q4QhhtW08QlWQSFTD8qQArVG4AIVsHJ2cVELgAGQ84CqqakHruZw9PMLgART8IeH7qwJA3QnzCLKsVKzgwy34PGAjzGHxkAA4ABil+CHyI214s-jhVoz6pAEcZ+GKZAxAVVAAWigcDgZjBlmInzkxGKKlK5SQlQWUnyITkPn8i1QYwm01mfSRA0WME2hzMx2QACYpL4VHIwmMAMIQEJlLJQaBvEB+fIAFU2BmRgyMAQAklBwbBnGBCHJZABBSXOGDpCbzfICARAA)

Reply:
handled